hw04/6.4JoesAutomative/Form1.cs

```diff
@@ -0,0 +1,133 @@
+namespace _6._4JoesAutomative
+{
+    public partial class Form1 : Form
+    {
+        private const decimal OIL_CHANGE = 50.00m;
+        private const decimal LUBE_JOB = 40.00m;
+        private const decimal RADIATOR_FLUSH = 100.00m;
+        private const decimal TRANSMISSION_FLUSH = 120.00m;
+        private const decimal INSPECTION = 35.00m;
+        private const decimal MUFFLER_REPLACEMENT = 150.00m;
+        private const decimal TIRE_ROTATION = 25.00m;
+        private const decimal LABOR_COST = 75.00m;
+        private const decimal TAX_RATE_ON_PARTS = 0.07m;
+        public Form1()
+        {
+            InitializeComponent();
+        }
+
+        private void Form1_Load(object sender, EventArgs e)
+        {
+
+        }
+
+        private decimal OilChangeCharges()
+        {
+            return cbOilChange.Checked ? OIL_CHANGE : 0m;
+        }
+
+        private decimal LubeCharges()
+        {
+            return cbLubeJob.Checked ? LUBE_JOB : 0m;
+        }
+
+        private decimal RadiatorFlushCharges()
+        {
+            return cbRadFlush.Checked ? RADIATOR_FLUSH : 0m;
+        }
+
+        private decimal TransmissionFlushCharges()
+        {
+            return cbTransFlush.Checked ? TRANSMISSION_FLUSH : 0m;
+        }
+
+        private decimal InspectionCharges()
+        {
+            return cbInspection.Checked ? INSPECTION : 0m;
+        }
+
+        private decimal MufflerCharges()
+        {
+            return cbRepMuff.Checked ? MUFFLER_REPLACEMENT : 0m;
+        }
+
+        private decimal TireRotationCharges()
+        {
+            return cbTireRot.Checked ? TIRE_ROTATION : 0m;
+        }
+
+        private decimal PartsCharges()
+        {
```

Comment:
Can i ask you why you used try-catch here?

Reply:
I use a try and catch block because anything other than a integer or decimal inputted in the textbox will raise an error. I catch these errors and return 0 instead, so the overall calculation is not affected. For example, if the input is "alim", an error would be raised because the string "alim" cannot be transformed into a decimal. The same applies for your next comment in the function LaborCharges.

Reply:
I see.
Do you think it'll be better if we show a messageBox saying that it's an invalid input in the catch phase?
I feel like as an user I would be confused if I get a 0 returned without telling me why.

Reply:
I tried doing that actually. I can look at it again. There was an issue with the btnCalc_Click function continuing to run because we catch the error, so the messageBox appears four times as we have four textboxes in summary.

Reply:
I think it's fine this way as everything stated in the textbook is implemented. If you really think it might cause an issue, I can look at it tomorrow (Tuesday).

Reply:
There is also a way to only accept numbers. such as..
 private void inputA_KeyPress(object sender, KeyPressEventArgs e)
 {
     if (!char.IsControl(e.KeyChar) && !char.IsDigit(e.KeyChar))
     {
         e.Handled = true;
     }
 }